src/subject.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2022 Snowplow Analytics Ltd. All rights reserved.
+//
+// This program is licensed to you under the Apache License Version 2.0,
+// and you may not use this file except in compliance with the Apache License Version 2.0.
+// You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the Apache License Version 2.0 is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+
+use derive_builder::Builder;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+/// Subject allows you to attach additional information about your application's environment
+///
+/// A Subject can be (attached to either a Tracker) where it will be sent with every Event, and/or
+/// attached to an Event, with the Event-level subject taking priority over Tracker-level
+#[derive(Serialize, Deserialize, Builder, Default, Clone, Debug)]
+#[builder(setter(into, strip_option), default)]
+pub struct Subject {
+    /// Unique identifier for user
+    #[serde(rename(serialize = "uid"))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub user_id: Option<String>,
+
+    /// Time zone of client device's OS
+    ///
+    /// This field is set automatically in enrichment if not present
+    #[serde(rename(serialize = "tz"))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timezone: Option<String>,
+
+    /// User language
+    #[serde(rename(serialize = "lang"))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub language: Option<String>,
+
+    /// User IP
+    ///
+    /// This field is set automatically in enrichment if not present
+    #[serde(rename(serialize = "ip"))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ip_address: Option<String>,
+
+    /// Useragent
+    ///
+    /// This field is set automatically in enrichment if not present
+    #[serde(rename(serialize = "ua"))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub useragent: Option<String>,
+
+    /// Domain-specific unique user identifier
+    #[serde(rename(serialize = "duid"))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub domain_uid: Option<String>,
+
+    /// Used to overwrite `nuid`
+    /// https://docs.snowplow.io/docs/collecting-data/collecting-from-own-applications/snowplow-tracker-protocol/#user-related-parameters
+    ///
+    /// This field is set automatically in enrichment if not present
+    #[serde(rename(serialize = "tnuid"))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub network_uid: Option<String>,
```

Comment:
Nit: also here `network_userid` is probably easier to understand.

Reply:
Or `network_user_id`? 🤔

Reply:
I prefer `network_user_id` I think. The other `_uid` fields have been updated as such also.